var/spack/repos/builtin/packages/phist/package.py

```diff
@@ -0,0 +1,149 @@
+for sparse linear and eigenvalue problems. In contrast to other libraries we
+support multiple backends (e.g. Trilinos, PETSc and our own optimized kernels),
+and interfaces in multiple languages such as C, C++, Fortran 2003 and Python.
+PHIST has a clear focus on portability and hardware performance: in particular
+we support row-major storage of block vectors and using GPUs (via the ghost
+library or Trilinos/Tpetra).
+"""
+
+    homepage = "https://bitbucket.org/essex/phist/"
+
+    version('develop',
+            git='https://bitbucket.org/essex/phist/phist.git', branch='devel')
+    version('master',
+            git='https://bitbucket.org/essex/phist/phist.git', branch='master')
+    version('1.4.3', 'af3300378d4282366d148e38c3a3199a',
+            url='https://bitbucket.org/essex/phist/get/phist-1.4.3.tar.gz')
+
+    # note: there is no virtual package for lapacke (the C bindings for
+    # lapack), We look for a file like lapacke.h in the lapack prefix, if it
+    # is found we pass it to phist's cmake, otherwise we hope that phist finds
+    #  it itself.
+    def lapacke_include_dir(self):
+        prefix = self.spec['lapack'].prefix
+        include_dir = "NOTFOUND"
+        for root, dir, files in os.walk(prefix):
+            if fnmatch.filter(files, "*lapacke.h"):
+                include_dir = root
+                break
+        return include_dir
+
+    def cmake_args(self):
+        spec = self.spec
+
+        kernel_lib = spec.variants['kernel_lib'].value
+        outlev = spec.variants['outlev'].value
+
+        lapacke_libs = \
+            (spec['lapack:c'].libs + spec['blas:c'].libs).joined(';')
+        lapacke_include_dir = format(spec['lapack:c'].headers.directories[0])
+
+        args = ['-DPHIST_KERNEL_LIB=%s' % kernel_lib,
+                '-DPHIST_OUTLEV=%s' % outlev,
+                '-DTPL_LAPACKE_LIBRARIES=%s' % lapacke_libs,
+                '-DTPL_LAPACKE_INCLUDE_DIRS=%s' % lapacke_include_dir,
+                '-DPHIST_ENABLE_MPI:BOOL=%s'
+                % ('ON' if '+mpi' in spec else 'OFF'),
+                '-DBUILD_SHARED_LIBS:BOOL=%s'
+                % ('ON' if '+shared' in spec else 'OFF'),
+                '-DPHIST_USE_TRILINOS_TPLS:BOOL=%s'
+                % ('ON' if '+trilinos' in spec else 'OFF'),
+                '-DPHIST_USE_SOLVER_TPLS:BOOL=%s'
+                % ('ON' if '+trilinos' in spec else 'OFF'),
+                '-DPHIST_USE_PRECON_TPLS:BOOL=%s'
+                % ('ON' if '+trilinos' in spec else 'OFF'),
+                ]
+
+        return args
+
+    @run_after('build')
+    @on_package_attributes(run_tests=True)
+    def check(self):
+        with working_dir(self.build_directory):
+            make("check")
+
+    @run_after('install')
+    @on_package_attributes(run_tests=True)
+    def test_install(self):
+        with working_dir(self.build_directory):
+            make("test_install")
+
+    variant(name='kernel_lib', default='builtin',
```

Comment:
could you please move `variants` on top? we usually have
```
versions()
variants()
depends_on()
<the-rest>
```

that's also because it's somewhat more natural to first get to know what variants are, then what are the dependencies, and then figure out how those are provided to the build system.